include/caffe/datum_DB.hpp

```diff
@@ -0,0 +1,142 @@
+#ifndef CAFFE_DATUMDB_HPP
+#define CAFFE_DATUMDB_HPP
+
+#include <string>
+
+#include "leveldb/db.h"
+#include "lmdb.h"
+
+#include "caffe/common.hpp"
+#include "caffe/datum_DB_factory.hpp"
+#include "caffe/proto/caffe.pb.h"
+
+namespace caffe {
+
+class DatumDBCursor {
+ public:
+  explicit DatumDBCursor(const DatumDBParameter& param)
+    : param_(param) {}
```

Comment:
I don't feel too strongly about this, but I think it could be argued that param passing should be left up to subclasses, and subclasses should be explicit about what information needs to be given to cursors. E.g., it seems that LMDB and LevelDB only need the single boolean `param_.loop()`, but doing it this way obscures that fact and makes one wonder if params might affect cursors in arbitrary ways.